src/libcore/cell.rs

```diff
@@ -1458,17 +1459,51 @@ impl<T: ?Sized + fmt::Display> fmt::Display for RefMut<'_, T> {
 /// ok (provided you enforce the invariants some other way), it is still undefined behavior
 /// to have multiple `&mut UnsafeCell<T>` aliases.
 ///
+///
 /// # Examples
 ///
+/// ## Wrapper not handing out references
+///
+/// A tiny wrapper similar to `Cell`, safe because it never creates an `&T` or `&mut T` reference:
+///
 /// ```
 /// use std::cell::UnsafeCell;
 ///
 /// # #[allow(dead_code)]
-/// struct NotThreadSafe<T> {
-///     value: UnsafeCell<T>,
+/// struct MyCell<T> {
+///     value: UnsafeCell<T>
+/// }
+///
+/// # #[allow(dead_code)]
+/// impl<T: Copy> MyCell<T> {
+///     fn get(&self) -> T {
+///         unsafe { *self.value.get() }
+///     }
+///
+///     fn set(&self, val: T) {
+///         unsafe { *self.value.get() = val}
+///     }
 /// }
+/// ```
+///
+/// Wrappers that hand out references get to complex for an example, where you somehow have to
+/// ensure there are no live references when modifying the wrapped value.
+///
+/// ## Using UnsafeCell through a raw reference
+///
+/// If you have a raw reference to an `UnsafeCell` that can't be turned into a shared reference
+/// (because for example the wrapped value may be uninitialized), use a pointer cast instead of
+/// `get`:
```

Comment:
I don't think we should bless this pattern yet, while the memory model is still so much in flux.

I like your other improvements, but for the interaction with `UnsafeCell` I'm afraid just changing docs won't cut it at this point.